File: backend/app/schemas/analytics.py
from typing import List

from pydantic import BaseModel


class SummaryMetrics(BaseModel):
    productivity_score: int
    completed_tasks: int
    pending_tasks: int
    total_tasks: int
    completion_rate: int
    focus_hours: float
    average_task_duration: float
    estimated_finish: str
    workload: str


class StreakMetrics(BaseModel):
    current: int
    longest: int


class WeeklyMetrics(BaseModel):
    labels: List[str]
    values: List[int]


class AnalyticsResponse(BaseModel):
    summary: SummaryMetrics

    streak: StreakMetrics

    weekly: WeeklyMetrics

    insights: List[str]

    recommendations: List[str]

File: frontend/src/types/analytics.ts
export interface SummaryMetrics {

    productivity_score: number;

    completed_tasks: number;

    pending_tasks: number;

    total_tasks: number;

    completion_rate: number;

    focus_hours: number;

    average_task_duration: number;

    estimated_finish: string;

    workload: string;

}

export interface StreakMetrics {

    current: number;

    longest: number;

}

export interface WeeklyMetrics {

    labels: string[];

    values: number[];

}

export interface AnalyticsResponse {

    summary: SummaryMetrics;

    streak: StreakMetrics;

    weekly: WeeklyMetrics;

    insights: string[];

    recommendations: string[];

}